src/app/scheduler/components/CourseCard.tsx

```diff
@@ -0,0 +1,72 @@
+/* eslint-disable prettier/prettier */
+import { AddIcon, CloseIcon, InfoOutlineIcon } from '@chakra-ui/icons';
+import { Box, Text, Flex, VStack, Checkbox, IconButton } from '@chakra-ui/react';
+import { useCallback } from 'react';
+
+import { useGetCourse, Term } from '../../../shared/fetchers';
+import { useSavedCourses } from '../../../shared/hooks/useSavedCourses';
+
+export type CourseCardProps = {
+    term: Term;
+    subject: string;
+    title: string;
+    code: string;
+    colour: string;
+    pid: string;
+    selected: boolean;
+    handleChange: Function;
+    handleInfo: Function;
+}
+
+export function CourseCard({ term, subject, title, code, colour, pid, handleChange, handleInfo, selected }: CourseCardProps): JSX.Element {
+    const { data } = useGetCourse({ term, pid: pid || '' })
```

Comment:
There's no need to use this function if you're passing in everything `data` offers as a prop. I'd pass `term` and `pid` as props, then get the rest of the information with this function fall. That'll clean up the props.